src/main/java/org/candlepin/insights/controller/InventoryController.java

```diff
@@ -262,6 +263,12 @@ private void extractVirtualizationFacts(Consumer consumer, Map<String, String> p
     private List<ConduitFacts> getValidatedConsumers(String orgId) {
         List<ConduitFacts> conduitFactsForOrg = new LinkedList<>();
         for (Consumer consumer : pinheadService.getOrganizationConsumers(orgId)) {
+            // Just call the whole thing off if the account number is missing
+            if (StringUtils.isEmpty(consumer.getAccountNumber())) {
+                log.warn("Account number is missing on consumers from {}", orgId);
+                return new ArrayList<>();
```

Comment:
Is it possible that only 1 (or a few) consumers returned from pinhead doesn't have an account number set? If so, I'm not sure that aborting the entire org update is correct since that single consumer would have to be fixed in order for the update to work. Thoughts?

Reply:
@mstead I was wondering the same thing, but @kahowell indicated that if one consumer were missing the accountNumber, they all would be.  @kahowell can you confirm I understood you correctly?

Reply:
Yeah, pinhead consults an internal service to get the accountNumber, so if one is missing it, they all will be.